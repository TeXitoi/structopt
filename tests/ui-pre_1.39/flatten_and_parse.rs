// Copyright 2018 Guillaume Pinot (@TeXitoi) <texitoi@texitoi.eu>
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

use structopt::StructOpt;

#[derive(StructOpt, Debug)]
struct DaemonOpts {
    #[structopt(short)]
    user: String,
    #[structopt(short)]
    group: String,
}

#[derive(StructOpt, Debug)]
#[structopt(name = "basic")]
struct Opt {
    #[structopt(flatten, parse(from_occurrences))]
    opts: DaemonOpts,
}

fn main() {
    let opt = Opt::from_args();
    println!("{:?}", opt);
}
