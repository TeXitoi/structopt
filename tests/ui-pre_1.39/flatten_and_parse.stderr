error: parse attribute is not allowed for flattened entry
  --> $DIR/flatten_and_parse.rs:22:26
   |
22 |     #[structopt(flatten, parse(from_occurrences))]
   |                          ^^^^^
